lib/trap.rb

```diff
@@ -26,10 +26,12 @@ def self.perform_trap(signal)
       Signal.trap(signal) do
         write_log("SIG#{signal}(*) received, ignoring...")
       end
-      @project&.handle_cancel
-      @project&.close
-      clean_threads
-      exit
+      Thread.start do
+        @project&.handle_cancel
+        @project&.close
+        clean_threads
+        exit
+      end
```

Comment:
I think we should wait for this thread. Or ruby will wait for all threads automatically at the exit?